src/rules/consentric-integration.js

```diff
@@ -0,0 +1,184 @@
+
+                const expiryInMs = new Date().getTime() + asMilliSeconds(expires_in);
+                const auth = {
+                    jwt: access_token,
+                    exp: expiryInMs
+                };
+
+                // Persist API Access token in global properties
+                global.consentricApiToken = auth;
+
+            } catch (error) {
+                console.error('Unable to retrieve API Access token for Consentric. Please check that your credentials (CONSENTRIC_CLIENT_ID and CONSENTRIC_CLIENT_SECRET) are correct.');
+                throw error;
+            }
+        }
+
+        return global.consentricApiToken;
+    };
+
+    // Creates Citizen Record in Consentric with Auth0 Id
+    const createCitizen = ({ userRef, apiAccessToken }) => {
+        console.log(`Upserting Consentric Citizen record for ${userRef}`);
+        const data = {
+            applicationId: CONSENTRIC_APPLICATION_ID,
+            externalRef: userRef,
+        };
+
+        return consentricApi
+            .post('/v1/citizens', data, {
+                headers: {
+                    Authorization: 'Bearer ' + apiAccessToken
+                },
+            })
+            .catch(err => {
+                if (err.response.status !== 409) { // 409 indicates Citizen with given reference already exists in Consentric
+                    console.error(err);
+                    throw err;
+                }
+            });
+    };
+
+    // Function to retrieve Consentric User Token from User Metadata
+    const getConsentricUserTokenFromMetadata = user => user.app_metadata && user.app_metadata.consentric;
+
+    // Generates On Demand Consentric User Token for the given User using the API Access Token
+    const generateConsentricUserAccessToken = async ({ userRef, apiAccessToken }) => {
+        try {
+            console.log(`Attempting to generate access token API for ${userRef}`);
+
+            const { data: { token, expiryDate: exp } } = await consentricApi
+                .post('/v1/access-tokens/tokens',
+                    {
+                        applicationId: CONSENTRIC_APPLICATION_ID,
+                        externalRef: userRef,
+                        expiryDate: moment().add(3, 'months').toISOString()
+                    },
+                    {
+                        headers: {
+                            Authorization: 'Bearer ' + apiAccessToken
+                        }
+                    }
+                );
+
+            return {
+                token,
+                exp
+            };
+
+        } catch (err) {
+            console.error(err);
+            throw err;
+        }
+    };
+
+    const loadConsentricUserAccessToken = async ({ user }) => {
+        try {
+            const metadataUserToken = getConsentricUserTokenFromMetadata(user);            
+            if ((metadataUserToken) && moment(metadataUserToken.exp).subtract(1, "days").isAfter(moment())) return metadataUserToken;
```

Comment:
That's quite a large skew :) Was that intentional? How long is the token expiration?

Reply:
The token is a 3 month expiry. I figured someone might sit with the page open for a while so it wouldn't hurt :-)